lib/kms/wrapper.js

```diff
@@ -0,0 +1,255 @@
+                            log.warn('error from kms',
+                                { implName, error: err });
+                            return next(err);
+                        }
+                        log.trace('data key ciphered by the kms');
+                        return next(null, cipheredDataKey);
+                    });
+            },
+            function createCipher(cipheredDataKey, next) {
+                log.debug('creating a cipher');
+                cipherBundle.cipheredDataKey =
+                    cipheredDataKey.toString('base64');
+                return Common.createCipher(cipherBundle.cryptoScheme,
+                    dataKey, 0, log, (err, cipher) => {
+                        if (err) {
+                            log.warn('error from kms',
+                            { implName, error: err });
+                            return next(err);
+                        }
+                        log.trace('cipher created by the kms');
+                        return next(null, cipher);
+                    });
+            },
+            function finishCipherBundle(cipher, next) {
+                cipherBundle.cipher = cipher;
+                return next(null, cipherBundle);
+            },
+        ], (err, cipherBundle) => {
+            if (err) {
+                log.error('error processing cipher bundle',
+                          { implName, error: err });
+            }
+            return cb(err, cipherBundle);
+        });
+    }
+
+     /**
+      * createDecipherBundle
+      * @param {object} serverSideEncryptionInfo - info for decryption
+      * @param {number} serverSideEncryptionInfo.cryptoScheme -
+      * cryptoScheme used
+      * @param {string} serverSideEncryptionInfo.algorithm -
+      * algorithm to use
+      * @param {string} serverSideEncryptionInfo.masterKeyId -
+      * key to get master key
+      * @param {boolean} serverSideEncryptionInfo.mandatory -
+      * true for mandatory encryption
+      * @param {buffer} serverSideEncryptionInfo.cipheredDataKey -
+      * ciphered data key
+      * @param {number} offset - offset for decryption
+      * @param {object} log - logger object
+      * @param {function} cb - cb from external call
+      * @returns {undefined}
+      * @callback called with (err, decipherBundle)
+      */
+    static createDecipherBundle(serverSideEncryptionInfo, offset,
+                                log, cb) {
+        if (!serverSideEncryptionInfo.masterKeyId ||
+            !serverSideEncryptionInfo.cipheredDataKey ||
+            !serverSideEncryptionInfo.cryptoScheme) {
+            log.error('Invalid cryptographic information', { implName });
+            return cb(errors.InternalError);
+        }
+        const decipherBundle = {
+            cryptoScheme: serverSideEncryptionInfo.cryptoScheme,
+            decipher: null,
+        };
+        return async.waterfall([
+            function decipherDataKey(next) {
```

Comment:
This function and the following ones are using parameters that are out of their own scope, it makes it hard to follow, and means we cannot rely on the function signature.